sdk/js/src/utils/parseVaa.ts

```diff
@@ -15,51 +14,37 @@ export const METADATA_REPLACE = new RegExp("\u0000", "g");
 //     131 [u8;len] uri
 //     ?   [u8; 32] recipient
 //     ?   u16      recipient_chain
-export const parseNFTPayload = (arr: Buffer) => {
-  const originAddress = arr.slice(1, 1 + 32).toString("hex");
-  const originChain = arr.readUInt16BE(33) as ChainId;
-  const symbol = Buffer.from(arr.slice(35, 35 + 32))
-    .toString("utf8")
-    .replace(METADATA_REPLACE, "");
-  const name = Buffer.from(arr.slice(67, 67 + 32))
-    .toString("utf8")
-    .replace(METADATA_REPLACE, "");
-  const tokenId = BigNumber.from(arr.slice(99, 99 + 32));
-  const uri_len = arr.readUInt8(131);
-  const uri = Buffer.from(arr.slice(132, 132 + uri_len))
-    .toString("utf8")
-    .replace(METADATA_REPLACE, "");
-  const target_offset = 132 + uri_len;
-  const targetAddress = arr
-    .slice(target_offset, target_offset + 32)
-    .toString("hex");
-  const targetChain = arr.readUInt16BE(target_offset + 32) as ChainId;
+export function parseNFTPayload(payload: Buffer): any {
+  const parsed = parseNftTransferPayload(payload);
   return {
-    originAddress,
-    originChain,
-    symbol,
-    name,
-    tokenId,
-    uri,
-    targetAddress,
-    targetChain,
+    originAddress: parsed.tokenAddress.toString("hex"),
+    originChain: parsed.tokenChain,
+    symbol: parsed.symbol,
+    name: parsed.name,
+    tokenId: BigNumber.from(parsed.tokenId),
+    uri: parsed.uri,
+    targetAddress: parsed.to.toString("hex"),
+    targetChain: parsed.toChain,
   };
-};
+}
 
 //     0   u256     amount
 //     32  [u8; 32] token_address
 //     64  u16      token_chain
 //     66  [u8; 32] recipient
 //     98  u16      recipient_chain
 //     100 u256     fee
-export const parseTransferPayload = (arr: Buffer) => ({
-  amount: BigNumber.from(arr.slice(1, 1 + 32)).toBigInt(),
-  originAddress: arr.slice(33, 33 + 32).toString("hex"),
-  originChain: arr.readUInt16BE(65) as ChainId,
-  targetAddress: arr.slice(67, 67 + 32).toString("hex"),
-  targetChain: arr.readUInt16BE(99) as ChainId,
-  fee: BigNumber.from(arr.slice(101, 101 + 32)).toBigInt(),
-});
+export function parseTransferPayload(payload: Buffer): any {
```

Comment:
I kept those as any because that is what they returned before. I really wanted to move these methods entirely, but I wasn't sure who else used these methods. So I kept these here and preserved the `any` return type.

Reply:
before the return type was unspecified which meant that it was inferred based on the definition, i.e. (copied from my editor):
```typescript
const parseTransferPayload: (arr: Buffer) => {
    amount: bigint;
    originAddress: string;
    originChain: ChainId;
    targetAddress: string;
    targetChain: ChainId;
    fee: bigint;
}
```

I think even if we don't add an explicit return type, we should just keep that behaviour

Reply:
Fixed. Also added `fromAddress` for the `parseTransferPayload` method because the old method would have assumed the from address is the fee for payload 3.